cmd/golangBuild.go

```diff
@@ -282,6 +282,11 @@ func runGolangBuild(config *golangBuildOptions, telemetryData *telemetry.CustomD
 			if !(response.StatusCode == 200 || response.StatusCode == 201) {
 				return fmt.Errorf("couldn't upload artifact, received status code %d", response.StatusCode)
 			}
+			// Publish list of binaries to cpe
+			if len(binaries) > 0 {
+				log.Entry().Infof("List of binaries %v\n", binaries)
+				commonPipelineEnvironment.custom.golangBinaries = binaries
```

Comment:
We've started to propose a "standard" for publishing result artifacts into the CPE with https://github.com/SAP/jenkins-library/pull/3717/files. It sounds like we should align this effort with the other.

Reply:
cc @OliverNocon

Reply:
The change here probably does not harm since it is used "internally" only.

Nonetheless, I agree that we need a consistent way of handling build results.
@anilkeshav27 could you maybe have a closer look?

Reply:
@OliverNocon I think I don't understand. How is this property of the CPE used "internally" in the `golangBuild` step only?

> The change here probably does not harm since it is used "internally" only.

If "internally" means anything other than "step internally", I disagree about it being internally.

Reply:
@loewenstein , currently there are 2 change that is needed for the internal download step of the golang artifact 

- the `groupId `from the versioning package that was missing in versioning but was considered in the `golangBuild` step 
- the golang binary name that only the the step knew and hence downloading the built artifact outside the step is currently not possible 

in general the current logic to identify the download url for a published artifact needs the repository base url that is that is then enriched with metadata (`groupdId`, `artifactId`, `version `and a few others) that is already identified in the `artifactPrepareVersionStep` and from the individual buill steps

Reply:
Thanks @anilkeshav27. If I get it correctly, this is pretty much in line what we want to align for Piper's artifact producing (i.e. build) steps and took a first shot on for `mavenBuild` in the PR I mentioned.

I strongly suggest to align this first, and then use a Piper "standard" way of communicating between steps via CPE about artifacts.

Reply:
@loewenstein , thanks for the information, I completely agree about the alignment

i had a close look into the PR https://github.com/SAP/jenkins-library/pull/3717/files

and writing the go binaries to the `cpe` is exactly what the artifacts concept from the above is pull is try to achieve. 

I see some discussions is still pending on the `artifact `object and also had a discussion with Oliver. Currently, `golangBuild` in Piper is failing since artifacts cannot be downloaded. In the interest of the pipeline that fail today, my suggestion is keep the current approach in PR as it and then once the PR 3717 is merged we can adapt the golangBuild / download to write / read the standard artifact cpe instead of the one currently implemented in this PR

what are your thoughts ?

Reply:
Resolving as https://github.com/SAP/jenkins-library/pull/3763 was raised and merged